toolchain/check/dump_id.h

```diff
@@ -0,0 +1,51 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_DUMP_ID_H_
+#define CARBON_TOOLCHAIN_CHECK_DUMP_ID_H_
+
+#include "toolchain/lex/token_kind.h"
+#include "toolchain/lex/tokenized_buffer.h"
+#include "toolchain/parse/tree.h"
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::Check {
+
+class Context;
+
+namespace DumpIdOverloads {
+
+auto DumpId(SemIR::LocId loc_id, const Context& context) -> void;
+
+}  // namespace DumpIdOverloads
+
+// A set of DumpId() overloads that dump an object to stderr, useful for calling
+// inside a debugger. These are all exposed as part of the `Check::Context`
+// API.
+//
+// This class is inherited by `Check::Context`, which provides itself as the
+// template parameter.
```

Comment:
Can you say a bit more about why this is implemented as a mixin instead of making these direct members of `Context`?

Reply:
Done.